### ISSUE TITLE
package ‘recount’ is not available for Bioconductor version '3.19'

### ISSUE THREAD

**Issue by twobeers75** [leekgroup/recount#25]
Hello,

Thanks in advance!

I'm having issues trying to install recount in R (Ubuntu 22.04 host) that I've not experienced before. Below is the terminal output and session info;

```
> if (!require("BiocManager", quietly = TRUE))
    install.packages("BiocManager")
Bioconductor version 3.19 (BiocManager 1.30.22), R 4.4.0 (2024-04-24)

> BiocManager::install("recount")
'getOption("repos")' replaces Bioconductor standard repositories, see
'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://cloud.r-project.org
Bioconductor version 3.19 (BiocManager 1.30.22), R 4.4.0 (2024-04-24)
Installing package(s) 'recount'
Warning message:
package ‘recount’ is not available for Bioconductor version '3.19'

A version of this package for your version of R might be available elsewhere,
see the ideas at
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages 

> sessionInfo()
R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Adelaide
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BiocManager_1.30.22

loaded via a namespace (and not attached):
[1] compiler_4.4.0 tools_4.4.0   
```

Package recount3 installs without issues as do many other packages however I still require recount.

As can be seen in the error, it has to do with the very recent release of the latest version of R and BiocManager. 

I am looking for any ideas about how I can overcome this incompatibility issue?

**Comment by MorfeoRenai** [comment]
Hullo, reporting basically the same issue of @twobeers75 

Recount3 installs without any issues for me as well.

I'll try doing some tests with older versions of R and BiocManager. If someone has any ideas I'll try to implement the and report results

```
> BiocManager::install('recount')
'getOption("repos")' replaces Bioconductor standard repositories,
see 'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://cran.rstudio.com/
Bioconductor version 3.19 (BiocManager 1.30.22), R 4.4.0
  (2024-04-24)
Installing package(s) 'recount'
Messaggio di avvertimento:
il pacchetto ‘recount’ non è disponibile for this version of R

Una versione di questo pacchetto per la tua versione di R potrebbe essere disponibile altrove,
si veda su
https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages 
> sessionInfo()
R version 4.4.0 (2024-04-24)
Platform: x86_64-pc-linux-gnu
Running under: ArcoLinux

Matrix products: default
BLAS/LAPACK: /usr/lib/libopenblas.so.0.3;  LAPACK version 3.12.0

locale:
 [1] LC_CTYPE=it_IT.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=it_IT.UTF-8        LC_COLLATE=it_IT.UTF-8    
 [5] LC_MONETARY=it_IT.UTF-8    LC_MESSAGES=it_IT.UTF-8   
 [7] LC_PAPER=it_IT.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=it_IT.UTF-8 LC_IDENTIFICATION=C       

time zone: Europe/Rome
tzcode source: system (glibc)

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

loaded via a namespace (and not attached):
[1] BiocManager_1.30.22 compiler_4.4.0      tools_4.4.0        
```

**Comment by MorfeoRenai** [comment]
OK `recount` seems to work with R versions `>=4.1` but `<4.4` obviously. To install several R versions I decided to use virtual environments: I used `micromamba` but using `mamba` or `conda` should give the same results.

A wierd thing that happens is that it requires to install R package `XML` (or maybe library `libxml2`) through micromamba/mamba/conda. After some testing I decided to go for the `4.3` version. 

``` bash
micromamba env create -n R-4-3-recount r-base=4.3 r-recommended r-xml
micromamba activate R-4-3-recount
# just to make sure we're using the correct R binary inside the virtual environment and the correct version
which R
R --version
# now run R
R
```

Now in R:

``` r
install.packages('BiocManager')
BiocManager::install('recount')
```

**Note**: if you're using RStudio, run RStudio from terminal after activating the virtual environment.

@twobeers75 let me know if it works for you!

**Comment by twobeers75** [comment]
Brilliant! I did not think this would work for my situation but I can confirm that it does. 

Thank you @MorfeoRenai, good enough work around for now. My code below;

I used Anaconda,
```
conda create -n r_env r-essentials r-base=4.3 r-recommended r-xml
conda activate r_env
```

then in R

```
R
>if (!require("BiocManager", quietly = TRUE))                                                                                                                                                              
    install.packages("BiocManager")
BiocManager::install('recount')
```

Works like a charm! Thanks again.

**Comment by lcolladotor** [comment]
Hi,

From https://bioconductor.org/packages/release/bioc/html/recount.html I do see the landing page for `recount` on BioC 3.19. Though there are some build errors in operating systems except for winOS https://bioconductor.org/checkResults/release/bioc-LATEST/recount/

This is is most likely related to issues with `rtracklayer::import.bw()` related to https://github.com/lawremi/rtracklayer/issues/73 and https://github.com/lawremi/rtracklayer/issues/83.

Best,
Leo

**Comment by lcolladotor** [comment]
Hi,

I posted an update at https://github.com/lawremi/rtracklayer/issues/83#issuecomment-2121313270 and also edited the `recount` vignette to implement a workaround to the `rtracklayer::import()` issues for remote BigWig files. The workaround involves downloading the data first.

Given the results at https://github.com/leekgroup/recount/actions/runs/9178757128, I anticipate that `recount3` will become available on bioc 3.19 and 3.20 on all platforms in 24-48 hrs.

I'll go ahead and close this issue in the meantime, but feel free to re-open it after 48 hrs if the issue persists.

Best,
Leo
